src/components/kalman_component/kalman.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+
+# Functions available in this file include:
+#   - setup_kf
+#   - run_kf
+
+# Description: Setup Kalman filter prereqiuisites
+# Usage:
+#   setup_kf
+setup_kf() {
+    StateVectorFile="${RunDirs}/StateVector.nc"
+
+    # Create a parent directory for the Kalman filter inversions
+    # Include a link to the state vector file for use with run_inversion.sh
+    mkdir -p ${RunDirs}/kf_inversions
+    ln -sf $StateVectorFile ${RunDirs}/kf_inversions/StateVector.nc
+
+    # copy kf notebook to kf_inversions directory
+    cp ${InversionPath}/src/notebooks/kf_notebook.ipynb ${RunDirs}/kf_inversions/
+
+    # Define Kalman filter update periods
+    python ${InversionPath}/src/components/kalman_component/make_periods_csv.py $StartDate $EndDate $UpdateFreqDays $RunDirs; wait
+
+    # Create unit scale factor file
+    python ${InversionPath}/src/components/kalman_component/make_unit_sf.py $StateVectorFile $RunDirs; wait
+
+    # Create directory to archive prior scale factors for each inversion period
+    mkdir -p ${RunDirs}/archive_sf
+
+    # Number of state vector elements
+    nElements=$(ncmax StateVector ${StateVectorFile})
+}
+
+# Description: Run Kalman filter inversions
+# Usage:
+#   run_kf
+run_kf() {
+
+    if ("$DoJacobian" && "$DoInversion" && "$DoPosterior"); then
+
+        # First run the Preview if necessary to get prior emissions
+        # needed for prepare_sf.py
+        if [[ ! -d ${RunDirs}/preview_run/OutputDir ]]; then
+            printf "\nPreview Dir not detected. Running the IMI Preview as a prerequisite for Kalman Mode.\n"
+            run_preview
```

Comment:
Can this component actually see `run_preview` ?

Reply:
I believe so. Think of the `source` command as pasting all the functions in the corresponding file that called `source kalman.sh`. Because `run_imi.sh` sources both preview.sh and kalman.sh, they should be able to see each other. I should probably check though